### PR TITLE
docs: fix documentation errors in data sources

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes the following error in data source:

```
====== Checking for data-sources huaweicloud_as_groups =====
the format of `groups.tags` is not correct, should be (Optional, Map)

====== Checking for data-sources huaweicloud_dcs_maintainwindow =====
the format of `end` is not correct, should be (Optional, String)
the format of `default` is not correct, should be (Optional, Bool)
the format of `seq` is not correct, should be (Optional, Int)

====== Checking for data-sources huaweicloud_dcs_instances =====
the format of `instances.tags` is not correct, should be (Optional, Map)

====== Checking for data-sources huaweicloud_dds_instances =====
the format of `instances.tags` is not correct, should be (Optional, Map)

====== Checking for data-sources huaweicloud_dms_rocketmq_broker =====
the format of `instance_id` is not correct, should be (Required, String, ForceNew)

====== Checking for data-sources huaweicloud_dms_kafka_instances =====
can not find `instances.retention_policy`, please check it

====== Checking for data-sources huaweicloud_dms_maintainwindow =====
the format of `default` is not correct, should be (Optional, Bool)
the format of `seq` is not correct, should be (Optional, Int)
the format of `end` is not correct, should be (Optional, String)

====== Checking for data-sources huaweicloud_dms_product =====
can not find `storage_spec_codes`, please check it

====== Checking for data-sources huaweicloud_images_images =====
can not find `image_type`, please check it
can not find `architecture`, please check it
can not find `images.image_type`, please check it
can not find `images.checksum`, please check it
can not find `images.updated_at`, please check it

====== Checking for data-sources huaweicloud_images_image =====
can not find `architecture`, please check it
can not find `image_type`, please check it
can not find `updated_at`, please check it

====== Checking for data-sources huaweicloud_iec_eips =====
can not find `eips.bandwidth_size`, please check it
can not find `eips.bandwidth_share_type`, please check it
can not find `eips.bandwidth_id`, please check it
can not find `eips.bandwidth_name`, please check it

====== Checking for data-sources huaweicloud_iec_network_acl =====
can not find `inbound_rules`, please check it
can not find `outbound_rules`, please check it
can not find `networks`, please check it
can not find `networks.vpc_id`, please check it
can not find `networks.subnet_id`, please check it

====== Checking for data-sources huaweicloud_iec_sites =====
can not find `region`, please check it

====== Checking for data-sources huaweicloud_iec_server =====
the format of `edgecloud_id` is not correct, should be (Optional, String)
can not find `image_id`, please check it
can not find `key_pair`, please check it
can not find `user_data`, please check it

====== Checking for data-sources huaweicloud_iec_vpc_subnets =====
the format of `subnets.dns_list` is not correct, should be (Optional, List)

====== Checking for data-sources huaweicloud_iec_vpc =====
can not find `subnet_num`, please check it

====== Checking for data-sources huaweicloud_identity_users =====
the format of `enabled` is not correct, should be (Optional, Bool)

====== Checking for data-sources huaweicloud_sfs_file_system =====
can not find `region`, please check it
can not find `state`, please check it
can not find `description`, please check it

====== Checking for data-sources huaweicloud_cbh_instances =====
can not find `instances.resource_id`, please check it

====== Checking for data-sources huaweicloud_cdm_flavors =====
the format of `region` is not correct, should be (Optional, String)

====== Checking for data-sources huaweicloud_cbr_vaults =====
the format of `vaults` is not correct, should be (Optional, List)

====== Checking for data-sources huaweicloud_gaussdb_opengauss_instances =====
can not find `instances`, please check it

====== Checking for data-sources huaweicloud_sms_source_servers =====
can not find `state`, please check it
can not find `servers.state`, please check it

====== Checking for data-sources huaweicloud_waf_dedicated_instances =====
can not find `instances`, please check it
can not find `instances.cpu_flavor`, please check it

====== Checking for data-sources huaweicloud_waf_instance_groups =====
can not find `groups.domain_names`, please check it
can not find `groups.domain_names.id`, please check it
can not find `groups.dedicated_instances`, please check it
can not find `groups.dedicated_instances.id`, please check it

====== Checking for data-sources huaweicloud_dws_flavors =====
can not find `flavors.elastic_volume_specs`, please check it

====== Checking for data-sources huaweicloud_modelarts_datasets =====
the format of `datasets.labels` is not correct, should be (Optional, List)

====== Checking for data-sources huaweicloud_vpc_route =====
the format of `nexthop` is not correct, should be (Optional, String)

====== Checking for data-sources huaweicloud_kms_data_key =====
the format of `plain_text` is not correct, should be (Optional, String)
the format of `cipher_text` is not correct, should be (Optional, String)
```

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/as" TESTARGS="-run TestAccDataSourceASGroup_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceASGroup_basic
=== PAUSE TestAccDataSourceASGroup_basic
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (148.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        148.939s

$ make testacc TEST="./huaweicloud/services/acceptance/waf" TESTARGS="-run TestAccDataSourceWafDedicatedInstances_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDataSourceWafDedicatedInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceWafDedicatedInstances_basic
=== PAUSE TestAccDataSourceWafDedicatedInstances_basic
=== CONT  TestAccDataSourceWafDedicatedInstances_basic
--- PASS: TestAccDataSourceWafDedicatedInstances_basic (359.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       359.874s

$ make testacc TEST="./huaweicloud/services/acceptance/waf" TESTARGS="-run TestAccDataSourceWafInstanceGroups_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDataSourceWafInstanceGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceWafInstanceGroups_basic
=== PAUSE TestAccDataSourceWafInstanceGroups_basic
=== CONT  TestAccDataSourceWafInstanceGroups_basic
--- PASS: TestAccDataSourceWafInstanceGroups_basic (455.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       455.977s
```

### CODE REVIEW

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 1]
```diff
@@ -247,7 +250,11 @@ func DataSourceASGroups() *schema.Resource {
 							Computed:    true,
 							Description: "The agency name.",
 						},
-						"tags": common.TagsSchema(),
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (deer-hang) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go:257]:
```suggestion
"tags": {
							Type:     schema.TypeMap,
							Computed: true,
							Elem:     &schema.Schema{Type: schema.TypeString},
						},
```

"tags": common.TagsComputedSchema(),

Comment (deer-hang) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]:
```suggestion
"tags": common.TagsComputedSchema(),
```

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go [hunk 2]
```diff
@@ -203,7 +203,11 @@ func DcsInstanceInstanceSchema() *schema.Resource {
 				Computed:    true,
 				Description: `Indicates the ID of the order that created the instance.`,
 			},
-			"tags": common.TagsSchema(),
+			"tags": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
```
Comment (deer-hang) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go]: ditto

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 3]
```diff
@@ -149,7 +149,11 @@ func DdsInstanceInstanceSchema() *schema.Resource {
 				Computed:    true,
 				Description: `Indicates the instance nodes information.`,
 			},
-			"tags": common.TagsSchema(),
+			"tags": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
```
Comment (deer-hang) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: ditto

File: docs/data-sources/iec_network_acl.md [hunk 4]
```diff
@@ -30,3 +30,5 @@ In addition to all arguments above, the following attributes are exported:
 
 * `description` - The description of the IEC network ACL.
 * `status` - The status of the IEC network ACL.
+* `inbound_rules` - A list of the IDs of ingress rules associated with the iec network ACL.
+* `outbound_rules` - A list of the IDs of egress rules associated with the iec network ACL.
```
Comment (Lance52259) [inline docs/data-sources/iec_network_acl.md]:
```suggestion
* `inbound_rules` - A list of the IDs of ingress rules associated with the IEC network ACL.
* `outbound_rules` - A list of the IDs of egress rules associated with the IEC network ACL.
```

File: docs/data-sources/sfs_file_system.md [hunk 5]
```diff
@@ -20,6 +20,9 @@ data "huaweicloud_sfs_file_system" "shared_file" {
 
 The following arguments are supported:
 
+* `region` - (Optional, String) Specifies the region in which to obtain the shared file systems.
```
Comment (Lance52259) [inline docs/data-sources/sfs_file_system.md]:
```suggestion
* `region` - (Optional, String) Specifies the region in which to obtain the shared file system.
```

**Pull request merged**